restorer.py

```diff
@@ -56,15 +56,15 @@ def restore_db_backup(self):
         Target:
             - restore a database's backup in PostgreSQL.
         '''
-        replicator = Replicator(self.connecter, self.new_dbname,
-                                Default.RESTORING_TEMPLATE, self.logger)
-        result = self.connecter.allow_db_conn(Default.RESTORING_TEMPLATE)
-        if result:
-            replicator.replicate_pg_db()
-            self.connecter.disallow_db_conn(Default.RESTORING_TEMPLATE)
-        else:
-            self.logger.stop_exe(Messenger.ALLOW_DB_CONN_FAIL.format(
-                dbname=Default.RESTORING_TEMPLATE))
+        #replicator = Replicator(self.connecter, self.new_dbname,
+                                #Default.RESTORING_TEMPLATE, self.logger)
+        #result = self.connecter.allow_db_conn(Default.RESTORING_TEMPLATE)
+        #if result:
+            #replicator.replicate_pg_db()
+            #self.connecter.disallow_db_conn(Default.RESTORING_TEMPLATE)
+        #else:
+            #self.logger.stop_exe(Messenger.ALLOW_DB_CONN_FAIL.format(
+                #dbname=Default.RESTORING_TEMPLATE))
```

Comment:
Those are the lines which create a new database before restoring the
selected backup (they call replicator because actually the new database is
a replica of template0).

Those lines are working, they create a database with the desired name, but
for any reason, despite being before the restoring process, the new
database isn't being recognized and the restorer stops the execution with
the error "The database X does not exist".

After the error, executing the restorer again will result in another error,
but this time in the replicator (cannot create the database with the
desired name because it already exists).

So, for the moment, I comment that code to avoid this problem, and make the
user create the database manually before restoring it.

2015-04-20 17:34 GMT+02:00 Alejandro Santana <notifications@github.com>:

> In restorer.py
> <https://github.com/anubia/py_pg_tools/pull/7#discussion_r28699235>:
>
> > -        result = self.connecter.allow_db_conn(Default.RESTORING_TEMPLATE)
> > -        if result:
> > -            replicator.replicate_pg_db()
> > -            self.connecter.disallow_db_conn(Default.RESTORING_TEMPLATE)
> > -        else:
> > -            self.logger.stop_exe(Messenger.ALLOW_DB_CONN_FAIL.format(
> > -                dbname=Default.RESTORING_TEMPLATE))
> > +        #replicator = Replicator(self.connecter, self.new_dbname,
> > +                                #Default.RESTORING_TEMPLATE, self.logger)
> > +        #result = self.connecter.allow_db_conn(Default.RESTORING_TEMPLATE)
> > +        #if result:
> > +            #replicator.replicate_pg_db()
> > +            #self.connecter.disallow_db_conn(Default.RESTORING_TEMPLATE)
> > +        #else:
> > +            #self.logger.stop_exe(Messenger.ALLOW_DB_CONN_FAIL.format(
> > +                #dbname=Default.RESTORING_TEMPLATE))
>
> Why comment these lines out?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anubia/py_pg_tools/pull/7/files#r28699235>.
>



-- 

--

*Juan Formoso Vasco*
jfv@anubia.es
· · · · · · · · · · · · · · · · · · · · · · · · · · · · ·

ANUBÍA, soluciones en la nube, S.L.

*Colón 26, planta 3, oficina 1*

36201, Vigo (Pontevedra)

www.anubia.es

[image: anubía_logo_2014_i_o_140x140.png][image:
odoo_logo_odoo_ready_140x140.png] <http://www.anubia.es/>